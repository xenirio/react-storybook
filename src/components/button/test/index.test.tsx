import React from 'react';
import { shallow } from 'enzyme';

import { Button, ButtonProps } from '../index';

const defaultProps: ButtonProps = {
    children: 'Click Me',
};

const setup = (props: Partial<ButtonProps> = {}) => {
    return shallow(<Button {...props} {...defaultProps} />);
};

describe('Button', () => {
    it('click', () => {
        const testOnClick = jest.fn();
        const component = setup({ onClick: testOnClick });
        component.simulate('click');
        expect(testOnClick).toHaveBeenCalledTimes(1);
    });

    it('render', () => {
        expect(setup().render()).toMatchSnapshot();
    });

    it('wide', () => {
        const component = setup({ wide: true });
        expect(component.hasClass('test-button--wide')).toBeTruthy();
    });

    it('className', () => {
        const customClassName = 'custom-class-123456';
        const component = setup({ className: customClassName });
        expect(component.hasClass(customClassName)).toBeTruthy();
    });
});
