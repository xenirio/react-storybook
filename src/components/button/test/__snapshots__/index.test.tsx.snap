// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render 1`] = `
<button
  class="test-button"
>
  Click Me
</button>
`;
